sim/riscv/sim-main.c

```diff
@@ -817,6 +818,21 @@ execute_i (SIM_CPU *cpu, unsigned_word iw, const struct riscv_opcode *op)
   unsigned_word shamt_imm = ((iw >> OP_SH_SHAMT) & OP_MASK_SHAMT);
   unsigned_word tmp;
   sim_cia pc = cpu->pc + 4;
+  host_callback *cb = STATE_CALLBACK (sd);
```

Comment:
This block of code should go inside the MATCH_ECALL case where is the only place it is used.  It is wasted work if we don't have a syscall.  You can add a brace level for the case MATCH_ECALL for the new local variables.